editor/editor_node.cpp

```diff
@@ -1324,22 +1324,31 @@ void EditorNode::save_resource_as(const Ref<Resource> &p_resource, const String
 		}
 	}
 
-	file->set_file_mode(EditorFileDialog::FILE_MODE_SAVE_FILE);
+	if (p_custom_file_dialog) {
```

Comment:
If you've done all this just to work around the issue where the file dialog appears behind your current dialog, then we have a system for auto-reparenting, which would be much simpler.

The dialog needs to be set up with `set_unparent_when_invisible(true)`, and then to pop it you'd use one of the `EditorInterface::get_singleton()->popup_dialog_*` methods.

Reply:
Unfortunately a lot of dialogs can't be popped up like that because they have custom functions to do it. I feel it may be better to just fix the issue properly. I will look into it.

Reply:
We can definitely extend this system for EditorFileDialog, since it's an editor-only node, without resolving it manually across all classes that need this.